data/ai/micro_ais/micro_ai_helper.lua

```diff
@@ -3,6 +3,56 @@ local MAIUV = wesnoth.require "ai/micro_ais/micro_ai_unit_variables.lua"
 
 local micro_ai_helper = {}
 
+local function is_number(value, check_int, ai_type, name)
+    -- Allow any number for 'float' type.
+    -- Allow anything that has an integer representation for 'integer', but
+    -- issue warning if it is given in 'float' notation.
+    local number = tonumber(value)
+    if (not number) then
+        return false
+    elseif check_int then
+        if (number ~= math.floor(number)) then
+            return false
+        elseif (math.type(number) ~= 'integer') then
+            std_print("Warning: [micro_ai] tag (" .. ai_type .. ") parameter '" .. name .. "' must be an integer. It has an integer representation, but is provided in floating-point format.")
```

Comment:
I think it might make sense to use `warning` instead of `std_print` here? I believe that adds it to the Lua console.